ui/app/adapters/secret-v2-version.js

```diff
@@ -70,14 +70,26 @@ export default ApplicationAdapter.extend({
 
   v2DeleteOperation(store, id, deleteType = 'delete') {
     let [backend, path, version] = JSON.parse(id);
-
-    // deleteType should be 'delete', 'destroy', 'undelete'
-    return this.ajax(this._url(backend, path, deleteType), 'POST', { data: { versions: [version] } }).then(
-      () => {
-        let model = store.peekRecord('secret-v2-version', id);
-        return model && model.rollbackAttributes() && model.reload();
-      }
-    );
+    // deleteType should be 'delete', 'destroy', 'undelete', 'delete-latest-version', 'destroy-version'
+    if ((!version && deleteType === 'delete') || deleteType === 'delete-latest-version') {
```

Comment:
before we were not handling the delete-latest-version which made it impossible for some people with proper permissions to execute this action.  They'd get a permissions denied, because we weren't actually deleting as those permissions allowed.

Reply:
We also weren't handling errors, so they would silently fail.